File: leetcode/algo2/day5/713_subarray_product_less_than_k.cpp
#include <vector>
#include <iostream>
int numSubarrayProductLessThanK(std::vector<int> &nums, int k)
{
    std::vector<int> debug;
    int res = 0;
    if (k <= 1)
        return 0;

    int left = 0;
    int product = 1;
    for (int right = 0; right < nums.size(); right++)
    {
        product *= nums[right];
        while (product >= k)
        {
            product /= nums[left++];
        }
        res += right - left + 1;
    }
    return res;
}
int main()
{
    std::vector<int> nums = {1, 1, 1};
    int k = 1;
    std::cout << numSubarrayProductLessThanK(nums, k) << std::endl;
}

File: leetcode/algo2/day5/713_subarray_product_less_than_k.js
var numSubarrayProductLessThanK = function (nums, k) {
  if (k <= 1) return 0;
  let left = 0;
  let product = 1;
  let res = 0;

  for (let right = 0; right < nums.length; right++) {
    product *= nums[right];
    while (product >= k) product /= nums[left++];
    res += right - left + 1;
  }
  return res;
};

let nums = [10, 5, 2, 6];
let k = 100;

console.log(numSubarrayProductLessThanK(nums, k));
